src/WizardBlock.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Collection\MapInterface;
+use Dhii\Event\EventFactoryInterface;
+use Dhii\Output\BlockInterface;
+use Dhii\Output\TemplateInterface;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\Modular\Events\EventsConsumerTrait;
+use stdClass;
+
+/**
+ * Class for showing wizard and enqueuing all assets that are related to it.
+ *
+ * @since [*next-version*]
+ */
+class WizardBlock implements BlockInterface
+{
+    /* @since [*next-version*] */
+    use EventsConsumerTrait;
+
+    /**
+     * Is assets attached already.
+     *
+     * @since [*next-version*]
+     *
+     * @var bool
+     */
+    protected static $isAssetsAttached = false;
+
+    /**
+     * Main application template.
+     *
+     * @since [*next-version*]
+     *
+     * @var TemplateInterface
+     */
+    protected $mainTemplate;
+
+    /**
+     * List of attributes for wizard.
+     *
+     * @since [*next-version*]
+     *
+     * @var array|MapInterface|stdClass
+     */
+    protected $attributes;
+
+    /**
+     * Rendered components templates.
+     *
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $renderedComponents;
+
+    /**
+     * WizardBlock constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param array|stdClass|MapInterface $attributes         List of attributes for wizard.
+     * @param TemplateInterface           $mainTemplate       Main application template.
+     * @param string                      $renderedComponents Rendered components templates.
```

Comment:
What does it mean for the component templates to be "rendered"? And why not accept `Stringable`?